homeassistant/components/system_bridge/__init__.py

```diff
@@ -200,6 +209,52 @@ def valid_device(device: str) -> str:
                 raise vol.Invalid(f"Could not find device {device}") from exception
         raise vol.Invalid(f"Device {device} does not exist")
 
+    async def handle_get_process_by_id(service_call: ServiceCall) -> ServiceResponse:
+        """Handle the get process by id service call."""
+        _LOGGER.debug("Get process by id: %s", service_call.data)
+        coordinator: SystemBridgeDataUpdateCoordinator = hass.data[DOMAIN][
+            service_call.data[CONF_BRIDGE]
+        ]
+        processes: list[Process] = coordinator.data.processes
+
+        # Find process.id from list, raise HomeAssistantError if not found
+        try:
+            return asdict(
+                next(
+                    process
+                    for process in processes
+                    if process.id == service_call.data[CONF_ID]
+                )
+            )
+        except StopIteration as exception:
+            raise HomeAssistantError(
```

Comment:
I think this should be a `ServiceValidationError`, because the user supplied an invalid process ID and there's no reason why we should spew a stack trace in the log when that happens.

Background in https://developers.home-assistant.io/blog/2023/11/30/service-exceptions-and-translations

Reply:
Changed 👍